### ISSUE TITLE
runSh job timeout?

### ISSUE THREAD

**Issue by jplimpens** [Shippable/support#3062]
I'd like to use Terrform to setup some infrastructure. Unfortunately it's not available in the pipeline (or at least I think it isn't). I'm not sure if the blog about Terraform is still valid (http://blog.shippable.com/infrastructure-provisioning-using-terraform)

My thoughts now are to run a custom job (runSh) after CI which creates/provisions some infrastructure (and later on destroys it). Therefor I created a script in my source repo. However, my runSh job can't find that script. The IN folder doesn't have these sources (not sure how to configure that).

Bypassing that script I use Task: - script: wget .. in the job to download Terraform (and extract it). However the runSh job finishes before the download is completed.

I think having Terraform available by default in the pipeline would be the way to go here, but the option of installing it in the job would be good enough.

Any thoughts on this which can point me in the right direction?

**Comment by avinci** [comment]
Thx for trying the product @jplimpens. We use terraform internally to manage everything. I can do a quick demo if you would like around this. Let me know what time works. Else, I can write a blog on how to do this...

**Comment by jplimpens** [comment]
cool. Either way would work for me. If not mistaken you guys are in PST. So what about a demo at 1100PST? That would be 2000CEST for me.

**Comment by manishas** [comment]
@jplimpens I will reach out to you and set up a demo for Friday or Monday, depending on your availability. Is jpmw.limpens@jp-l.org the right email to reach you at?

In the meantime, we just published a blog on provisioning infrastructure with terraform using a runSh job : [Provisioning AWS infrastructure with Terraform](http://blog.shippable.com/provisioning-aws-infrastructure-with-terraform). The sample project should give you some pointers.

**Comment by jplimpens** [comment]
Yep. that's the correct email. Friday would work best for me.

**Comment by jplimpens** [comment]
FYI: based on your example code and blog I've got it working (sort of) in my 'HelloWorld' project. I've run into an authorization failure (not authorized for the operation). However, I've fixed my main issues (the script runs and terraform download etc works). I'll fix the auth error tomorrow. Still some questions left (timing/run-order issues between repo rsync & custom job and the difference between seeded repo and resource repo - since these are actually the same) , which hopefully I will understand better after demo. 

Much obliged
